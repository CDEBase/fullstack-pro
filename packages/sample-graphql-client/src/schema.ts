/* tslint:disable */
//  This file was automatically generated and should not be edited.

export type addCountMutationVariables = {
  amount: number,
};

export type addCountMutation = {
  // Increase counter value, returns current counter amount
  addCount:  {
    // Current amount
    amount: number,
  } | null,
};

export type addPersonMutationVariables = {
  name: string,
  sex: string,
};

export type addPersonMutation = {
  addPerson:  {
    id: string | null,
    name: string | null,
  } | null,
};

export type getCountQuery = {
  // Counter
  count:  {
    // Current amount
    amount: number,
  } | null,
};

export type getPersonQueryVariables = {
  id: string,
};

export type getPersonQuery = {
  getPerson:  {
    id: string | null,
    name: string | null,
    sex: string | null,
    matches:  Array< {
      id: string | null,
      name: string | null,
      sex: string | null,
      matches:  Array< {
        id: string | null,
        name: string | null,
        sex: string | null,
        matches:  Array< {
          id: string | null,
          name: string | null,
          sex: string | null,
        } | null > | null,
      } | null > | null,
    } | null > | null,
  } | null,
};

export type getPersonsQuery = {
  persons:  Array< {
    id: string | null,
    name: string | null,
  } | null > | null,
};

export type subscribeToCountSubscription = {
  // Subscription fired when anyone increases counter
  subscribeToCount:  {
    // Current amount
    amount: number,
  } | null,
};
/* tslint:enable */
