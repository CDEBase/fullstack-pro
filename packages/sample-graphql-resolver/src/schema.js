// @flow

import {makeExecutableSchema, addMockFunctionsToSchema} from 'graphql-tools';

export const queries = `
  me: User
  user(id: ID!): User
`;

export const mutations = `
  createAccount(user: UserPasswordInput!): User
  createAccountAndLogin(user: UserPasswordInput!): Token
  loginWithPassword(user: UserPasswordInput!): Token
`;

export const typeDefs = `
  type Token {
    userId: ID!
    token: String!
    tokenExpiration: Float!
  }
  
  input UserPasswordInput {
    email: String
    username: String
    password: String
  }
  type User {
    id: ID!
    email: String
    username: String
  }
`;

export const rootObjectsExtension = `
  extend type RootQuery {
    ${queries}
  }
  extend type RootMutation {
    ${mutations}
  }
`;

const schema = `
  type RootQuery {
    ${queries}
  }
  type RootMutation {
    ${mutations}
  }
  schema {
    query: RootQuery
    mutation: RootMutation
  }
`;

const exectuableSchema = makeExecutableSchema({ typeDefs: [schema, typeDefs] });

addMockFunctionsToSchema({schema: exectuableSchema});

export default exectuableSchema;