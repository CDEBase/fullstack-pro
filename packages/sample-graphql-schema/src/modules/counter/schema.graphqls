# Database counter
type Count {
    # Current amount
    amount: Int!
}

extend type Query {
    # Counter 
    count: Count
}

extend type Mutation {
    # Increase counter value, returns current counter amount
    addCount(
        # Ammount to add to counter
        amount: Int!
    ): Count
}

extend type Subscription {
    # Subscription fired when anyone increases counter
    subscribeToCount: Count
}