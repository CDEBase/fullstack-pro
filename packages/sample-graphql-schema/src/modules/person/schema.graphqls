# Person Type
type PersonType {
    name: String
    id: String
    sex: String
    matches: [PersonType]
}

extend type Query {
    getPerson(id: String!): PersonType
    persons: [PersonType]
}

extend type Mutation {
    addPerson(name: String!, sex: String!): PersonType
}

extend type Subscription {
    # Subscription fires on every comment added
    clock(onlyMinutesChange: Boolean): String
}
