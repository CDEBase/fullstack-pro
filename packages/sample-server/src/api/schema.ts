import {GraphQLSchema} from 'graphql';
import {makeExecutableSchema, addMockFunctionsToSchema, addErrorLoggingToSchema} from 'graphql-tools';
import { resolvers, typeDefs } from '@sample/schema';
const rootSchemaDef = require('./root_schema.graphqls');
import { logger } from '../../../../src/logger';

const schema: GraphQLSchema = makeExecutableSchema({
  logger: console,
  resolverValidationOptions: {
    requireResolversForNonScalar: false,
  },
  resolvers: resolvers,
  typeDefs: [rootSchemaDef].concat(typeDefs),
});

addErrorLoggingToSchema(schema, { log: (e) => logger.error(e) });

addMockFunctionsToSchema({
  mocks: {},
  preserveResolvers: true,
  schema,
});


export {schema};
