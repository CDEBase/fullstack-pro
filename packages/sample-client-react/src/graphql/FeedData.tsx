import { graphql } from 'react-apollo';

