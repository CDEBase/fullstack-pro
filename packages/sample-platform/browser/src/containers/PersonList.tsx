import * as React from 'react';
import { graphql } from 'react-apollo';
import compose from 'lodash/flowRight';
import { PERSONS_QUERY, getPersonsQuery } from '@sample-stack/platform-browser';

export interface IPersonListProps {
    persons;
}
const PersonListComponent: React.SFC<IPersonListProps> = ({ persons }) => (
    <div>
        <h2>Persons:</h2>
       {persons && persons.map((person, i) => <div key={i}>{person.name}</div>)}
    </div>
);

export const PersonList: React.ComponentClass<{}> =
    compose(
        graphql<{}, getPersonsQuery, {}, {}>(PERSONS_QUERY),
        // flattenProp('data'),
    )(PersonListComponent);
