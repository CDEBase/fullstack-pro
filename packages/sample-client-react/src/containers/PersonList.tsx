import * as React from 'react';
import { graphql, gql } from 'react-apollo';
import { pure, compose, flattenProp } from 'recompose';
import { getPersons } from '@sample/graphql';

// const getPersons = gql`query getPersons {
//     persons {
//         id
//         name
//     }
// }`
export interface IPersonListProps {
    persons
}
const PersonListComponent: React.SFC<IPersonListProps> = ({ persons }) => (
    <div>
        <h2>Persons:</h2>
        {console.log(persons)} {persons && persons.map((person, i) => <div key={i}>{person.name}</div>)}
    </div>
);

export const PersonList: React.ComponentClass<{}> =
    compose(
        graphql(getPersons),
        flattenProp('data'),
        pure
    )(PersonListComponent);