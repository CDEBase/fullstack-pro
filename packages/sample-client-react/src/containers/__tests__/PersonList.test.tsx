import 'jest';
import * as React from 'react';
import { shallow, mount } from 'enzyme';
import ApolloClient from 'apollo-client';
import { MockedProvider } from 'react-apollo/lib/test-utils';
import './setup';

declare function require(name: string);
import * as TestUtils from 'react-addons-test-utils';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import gql from 'graphql-tag';

import { PersonList } from '../PersonList';
import { reducers, Store } from '@sample/client-redux';
import { ApolloProvider, graphql, createNetworkInterface } from 'react-apollo';
import { database } from '@sample/schema';
import { mockNetworkInterface } from 'react-apollo/lib/test-utils';
import { PERSONS_QUERY, GetPersonsQuery } from '@sample/graphql';

describe('components/PersonList', () => {
  it('renders correctly', (done) => {
    const networkInterface = mockNetworkInterface({
      request: { query: PERSONS_QUERY, variables: {} }, result: { data: database.persons },
    });

    const client = new ApolloClient({
      networkInterface: networkInterface,
    });


    const store = createStore(combineReducers({
      ...reducers,
      apollo: client.reducer(),
    }), applyMiddleware(client.middleware()));


    expect(mount(
      <ApolloProvider client={client} store={store}>
        <PersonList />
      </ApolloProvider>,
    )).toMatchSnapshot();
  });
});
