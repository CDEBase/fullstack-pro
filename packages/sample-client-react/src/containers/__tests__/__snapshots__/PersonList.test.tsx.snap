// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PersonList renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ApolloProvider
    client={
        ApolloClient {
            "addTypename": true,
            "dataId": [Function],
            "dataIdFromObject": [Function],
            "disableNetworkFetches": false,
            "fieldWithArgs": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "initialState": Object {},
            "middleware": [Function],
            "mutate": [Function],
            "networkInterface": MockNetworkInterface {
              "mockedResponsesByKey": Object {},
            },
            "query": [Function],
            "queryDeduplication": true,
            "reducerConfig": Object {
              "addTypename": true,
              "customResolvers": undefined,
              "dataIdFromObject": [Function],
              "fragmentMatcher": [Function],
            },
            "resetStore": [Function],
            "setStore": [Function],
            "ssrMode": false,
            "version": "local",
            "watchQuery": [Function],
          }
    }
    store={null}
>
    <Apollo(flattenProp(pure(PersonListComponent))) />
</ApolloProvider>,
  "nodes": Array [
    <ApolloProvider
      client={
            ApolloClient {
                  "addTypename": true,
                  "dataId": [Function],
                  "dataIdFromObject": [Function],
                  "disableNetworkFetches": false,
                  "fieldWithArgs": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "initialState": Object {},
                  "middleware": [Function],
                  "mutate": [Function],
                  "networkInterface": MockNetworkInterface {
                    "mockedResponsesByKey": Object {},
                  },
                  "query": [Function],
                  "queryDeduplication": true,
                  "reducerConfig": Object {
                    "addTypename": true,
                    "customResolvers": undefined,
                    "dataIdFromObject": [Function],
                    "fragmentMatcher": [Function],
                  },
                  "resetStore": [Function],
                  "setStore": [Function],
                  "ssrMode": false,
                  "version": "local",
                  "watchQuery": [Function],
                }
      }
      store={null}
>
      <Apollo(flattenProp(pure(PersonListComponent))) />
</ApolloProvider>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <MockedProvider>
        <Apollo(flattenProp(pure(PersonListComponent))) />
</MockedProvider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": MockedProvider {
        "_reactInternalInstance": [Circular],
        "client": ApolloClient {
          "addTypename": true,
          "dataId": [Function],
          "dataIdFromObject": [Function],
          "disableNetworkFetches": false,
          "fieldWithArgs": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "initialState": Object {},
          "middleware": [Function],
          "mutate": [Function],
          "networkInterface": MockNetworkInterface {
            "mockedResponsesByKey": Object {},
          },
          "query": [Function],
          "queryDeduplication": true,
          "reducerConfig": Object {
            "addTypename": true,
            "customResolvers": undefined,
            "dataIdFromObject": [Function],
            "fragmentMatcher": [Function],
          },
          "resetStore": [Function],
          "setStore": [Function],
          "ssrMode": false,
          "version": "local",
          "watchQuery": [Function],
        },
        "context": Object {},
        "props": Object {
          "children": <Apollo(flattenProp(pure(PersonListComponent))) />,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ApolloProvider
          client={
                    ApolloClient {
                              "addTypename": true,
                              "dataId": [Function],
                              "dataIdFromObject": [Function],
                              "disableNetworkFetches": false,
                              "fieldWithArgs": [Function],
                              "fragmentMatcher": HeuristicFragmentMatcher {},
                              "initialState": Object {},
                              "middleware": [Function],
                              "mutate": [Function],
                              "networkInterface": MockNetworkInterface {
                                "mockedResponsesByKey": Object {},
                              },
                              "query": [Function],
                              "queryDeduplication": true,
                              "reducerConfig": Object {
                                "addTypename": true,
                                "customResolvers": undefined,
                                "dataIdFromObject": [Function],
                                "fragmentMatcher": [Function],
                              },
                              "resetStore": [Function],
                              "setStore": [Function],
                              "ssrMode": false,
                              "version": "local",
                              "watchQuery": [Function],
                            }
          }
          store={null}
>
          <Apollo(flattenProp(pure(PersonListComponent))) />
</ApolloProvider>,
        "_debugID": 2,
        "_renderedOutput": <ApolloProvider
          client={
                    ApolloClient {
                              "addTypename": true,
                              "dataId": [Function],
                              "dataIdFromObject": [Function],
                              "disableNetworkFetches": false,
                              "fieldWithArgs": [Function],
                              "fragmentMatcher": HeuristicFragmentMatcher {},
                              "initialState": Object {},
                              "middleware": [Function],
                              "mutate": [Function],
                              "networkInterface": MockNetworkInterface {
                                "mockedResponsesByKey": Object {},
                              },
                              "query": [Function],
                              "queryDeduplication": true,
                              "reducerConfig": Object {
                                "addTypename": true,
                                "customResolvers": undefined,
                                "dataIdFromObject": [Function],
                                "fragmentMatcher": [Function],
                              },
                              "resetStore": [Function],
                              "setStore": [Function],
                              "ssrMode": false,
                              "version": "local",
                              "watchQuery": [Function],
                            }
          }
          store={null}
>
          <Apollo(flattenProp(pure(PersonListComponent))) />
</ApolloProvider>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MockedProvider>
    <Apollo(flattenProp(pure(PersonListComponent))) />
</MockedProvider>,
}
`;
